network/basics/uri.py

```diff
@@ -415,8 +449,20 @@ def main():
         if not os.path.exists(removes):
             module.exit_json(stdout="skipped, since %s does not exist" % removes, changed=False, stderr=False, rc=0)
 
+    if source is not None:
+        # do not run the command if the line contains source=filename
+        # and the filename do not exists.  This allows idempotence
+        # of uri executions.
+        if not os.path.exists(source) or not os.access(source, os.R_OK):
+            module.exit_json(stdout="skipped, since %s does not exist or is unaccessible" % source, changed=False, stderr=False, rc=0)
+        if  not ( method == 'PUT' or method == 'POST'):
+            module.exit_json(stdout="skipped, since %s is not allowed method for sending binary data" % method, changed=False, stderr=False, rc=0)
```

Comment:
If we can't do this then it should be an error.  I don't, however, see that we currently do this for body so I'm not sure if it should be an error or we should allow trying to send it even without PUT/POST then return an error when making the request if there's an issue.